### PR TITLE
updated typescript and added correct type generation for all typescript files

### PR DESCRIPTION
**Pull request by jeudy100**

### CODE REVIEW

**Comment by jeudy100** [comment]
@anujsi please review

**Comment by jeudy100** [comment]
@anujsi I couldn't update webpack because the html-webpack-inline-source-plugin hasn't been updated in 4 years and doesn't work with the newest version.

File: package.json [hunk 1]
```diff
@@ -3,11 +3,21 @@
...
+    "dist/*",
+    "dist/**/*",
+    "@types/**"
+  ],
+  "directories": {
+    "lib": "dist",
+    "types": "@types"
+  },
   "scripts": {
-    "clean": "rimraf dist",
```
Comment (anujsingla) [inline package.json:8]: you can write like this: "clean": "rimraf dist @types",

**Comment by anujsingla** [comment]
Just one comment. Apart from this looks good to me

**Pull request merged**